geomeffibem/vertex.py

```diff
@@ -47,6 +51,91 @@ def get_coords_on_plane(self, plane='xy') -> Tuple[float, float]:
+            mult = newLength / currentLength
+            self.x *= mult
+            self.y *= mult
+            self.z *= mult
+        else:
+            raise ValueError("Cannot normalize a vector of length 0")
+
+    def dot(self, other) -> float:
+        """Computes the dot / scalar / inner product product of two vectors.
+
+        (a.b).
+        """
+        # return np.dot(v.to_numpy(), v2.to_numpy())
+        return self.x * other.x + self.y * other.y + self.z * other.z
+
+    def cross(self, other, normalize: bool = False) -> Vertex:
+        """Computes the cross product (a x b), which is a vector perpendicular to both a and b."""
+        v = Vertex(
+            x=(self.y * other.z - self.z * other.y),
+            y=(self.z * other.x - self.x * other.z),
+            z=(self.x * other.y - self.y * other.x),
+        )
+        if normalize:
+            return v.normalize()
+        return v
+
+    def outer_product(self, other) -> np.ndarray:
+        """Compute the outer product of this by another vector."""
+        return np.outer(self.to_numpy(), other.to_numpy())
+
+    def __add__(self, other) -> Vertex:
+        """Return a + b."""
+        return Vertex(x=self.x + other.x, y=self.y + other.y, z=self.z + other.z)
+
+    def __neg__(self):
+        """Return obj negated (-obj)."""
+        return Vertex(-self.x, -self.y, -self.z)
+
+    def __sub__(self, other) -> Vertex:
+        """Return a - b."""
+        return self + -other
+        # return Vertex(
+        #     x=self.x - other.x,
+        #     y=self.y - other.y,
+        #     z=self.z - other.z
+        # )
+
+    def __mul__(self, other) -> Vertex:
+        """Multiplies each coordinate by a scalar."""
+        if not isinstance(other, int) and not isinstance(other, float):
+            raise ValueError("Multiplication of a vertex by something else than a numeric is not supported")
+        return Vertex(self.x * other, self.y * other, self.z * other)
+
+    def __rmul__(self, other) -> Vertex:
+        """Multiplies each coordinate by a scalar."""
+        if not isinstance(other, int) and not isinstance(other, float):
+            raise ValueError("Multiplication of a vertex by something else than a numeric is not supported")
+        return self.__mul__(other)
+
+    def __truediv__(self, other) -> Vertex:
+        """Divides each coordinate by a scalar."""
+        if not isinstance(other, int) and not isinstance(other, float):
+            raise ValueError("Division of a vertex by something else than a numeric is not supported")
+        return Vertex(self.x / other, self.y / other, self.z / other)
+
+    def __floordiv__(self, other) -> Vertex:
+        """Divides each coordinate by a scalar."""
+        if not isinstance(other, int) and not isinstance(other, float):
+            raise ValueError("Division of a vertex by something else than a numeric is not supported")
+        return Vertex(self.x // other, self.y // other, self.z // other)
```

Comment:
Extend the Vertex class with dot/cross products, and muplication / division / addition